### PR TITLE
[`PEFT` / `Tests`] Add peft slow tests on push

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

As per title.

Currently PEFT slow tests are not being tested on main push. This PR adds a new job in the slow tests workflow to make sure we test PEFT slow tests on main 

cc @patrickvonplaten @DN6 @sayakpaul 

I am also happy to push this change on a dummy branch on hf repo to test the implementation

### CODE REVIEW

File: .github/workflows/push_tests.yml [hunk 1]
```diff
@@ -156,6 +156,56 @@ jobs:
...
+
+    - name: Run slow PEFT CUDA tests
+      env:
+        HUGGING_FACE_HUB_TOKEN: ${{ secrets.HUGGING_FACE_HUB_TOKEN }}
+        # https://pytorch.org/docs/stable/notes/randomness.html#avoiding-nondeterministic-algorithms
+        CUBLAS_WORKSPACE_CONFIG: :16:8
+      run: |
+        python -m pytest -n 1 --max-worker-restart=0 --dist=loadfile \
+          -s -v -k "not Flax and not Onnx" \
+          --make-reports=tests_torch_cuda \
```
Comment (younesbelkada) [inline .github/workflows/push_tests.yml:193]:
```suggestion
          --make-reports=tests_peft_cuda \
```

**Comment by patrickvonplaten** [comment]
@DN6 can you check here?

**Comment by sayakpaul** [comment]
@DN6 gentle ping.

File: .github/workflows/push_tests.yml [hunk 2]
```diff
@@ -156,6 +156,56 @@ jobs:
...
+    - name: Run slow PEFT CUDA tests
+      env:
+        HUGGING_FACE_HUB_TOKEN: ${{ secrets.HUGGING_FACE_HUB_TOKEN }}
+        # https://pytorch.org/docs/stable/notes/randomness.html#avoiding-nondeterministic-algorithms
+        CUBLAS_WORKSPACE_CONFIG: :16:8
+      run: |
+        python -m pytest -n 1 --max-worker-restart=0 --dist=loadfile \
+          -s -v -k "not Flax and not Onnx" \
+          --make-reports=tests_peft_cuda \
+          tests/lora/
```
Comment (sayakpaul) [inline .github/workflows/push_tests.yml:194]: Under `lora` we have the test suite that doesn't utilize `peft`. Do we wanna include them here?

**Review by sayakpaul — APPROVED** [review]
Thanks, this works for me!

File: .github/workflows/push_tests.yml (hunk 2 above)
Comment (younesbelkada) [reply .github/workflows/push_tests.yml:194]: Yes but they're going to be skipped thanks to this decorator: https://github.com/huggingface/diffusers/blob/main/tests/lora/test_lora_layers_old_backend.py#L1883 being set everywhere

**Comment by patrickvonplaten** [comment]
Reping @dhruvrnaik

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**